docs/src/getting_started/optim_compatibility.md

```diff
@@ -0,0 +1,188 @@
+
+```julia
+light_gt = PointLight(
+    Vec3(1.0f0, 1.0f0, 1.0f0),
+    20000.0f0,
+    Vec3(1.0f0, 10.0f0, -50.0f0)
+)
+
+target_img = render(light_gt, scene)
+
+showimg(zeroonenorm(render(light_gt, scene)))
+```
+
+```@raw html
+<p align="center">
+    <img width=300 height=300 src="../../assets/inv_light_original.png">
+</p>
+```
+
+## Initial Guess of Lighting Parameters
+
+We shall make some arbitrary guess of the lighting parameters (intensity and
+position) and try to get back the image in [Ground Truth Image](@ref optim)
+
+```julia
+light_guess = PointLight(
+    Vec3(1.0f0, 1.0f0, 1.0f0),
+    1.0f0,
+    Vec3(-1.0f0, -10.0f0, -50.0f0)
+)
+
+showimg(zeroonenorm(render(light_guess, scene)))
+```
+
+```@raw html
+<p align="center">
+    <img width=300 height=300 src="../../assets/inv_light_initial.png">
+</p>
+```
+
+## Writing the Optimization Loop using Optim
+
+Since, there is no direct support of Optim (unlike for Flux) in RayTracer
+the interface might seem a bit ugly. This is mainly due to the way the
+two optimization packages work. Flux allows inplace operation and ideally
+even RayTracer prefers that. But Optim requires us to give the parameters
+as an `AbstractArray`.
+
+Firstly, we shall extract the parameters, using the [`RayTracer.get_params`](@ref)
+function, we want to optimize.
+
+```julia
+initial_parameters = RayTracer.get_params(light_guess)[end-3:end]
+```
+
+Since the input to the `loss_function` is an abstract array we need to
+convert it into a form that the RayTracer understands. For this we
+shall use the [`RayTracer.set_params!`](@ref) function which will modify the
+parameters inplace.
+
+In this function we simply compute the loss values and print it for our
+reference
+
+```julia
+function loss_function(θ::AbstractArray)
+    light_optim = deepcopy(light_guess)
+    RayTracer.set_params!(light_optim.intensity, θ[1:1])
+    RayTracer.set_params!(light_optim.position, θ[2:end])
+    loss = sum((render(light_optim, scene) .- target_img) .^ 2)
+    @show loss
+    return loss
+end
+```
+
+RayTracer uses Zygote's Reverse Mode AD for computing the derivatives.
+However, the default in Optim is ForwardDiff. Hence, we need to override
+that by giving our own gradient function.
+
+```julia
+function ∇loss_function!(G, θ::AbstractArray)
+    light_optim = deepcopy(light_guess)
+    RayTracer.set_params!(light_optim.intensity, θ[1:1])
+    RayTracer.set_params!(light_optim.position, θ[2:end])
+    gs = gradient(light_optim) do L
+        sum((render(L, scene) .- target_img) .^ 2)
+    end
+    G .= RayTracer.get_params(gs[1])[end-3:end]
+end
+```
+
+Now we simply call the `optimize` function with `LBFGS` optimizer.
+
+```julia
+res = optimize(loss_function, ∇loss_function!, initial_parameters, LBFGS())
+
+@show res.minimizer
+```
+
+One interesting thing to notice is that LBFGS took us to the global minima while
+ADAM in the previous tutorial was only able to reach a local minima. Also, the
```

Comment:
Are you sure BFGS got the exact global minimum, and not just a better solution than Adam "by chance"? There is no reason for BFGS to reach the global optimum, as the function you are minimising is not convex.

Reply:
Yes it got it by chance. Maybe I should rephrase the sentence.